roofit/roofitcore/test/TestStatistics/testLikelihoodGradientJob.cpp

```diff
@@ -355,20 +369,30 @@ TEST_F(LikelihoodSimBinnedConstrainedTest, BasicParameters)
    EXPECT_DOUBLE_EQ(nll0, nll1);
 }
 
-TEST_F(LikelihoodSimBinnedConstrainedTest, ConstrainedAndOffset)
+class SimBinnedConstrainedTest : public ::testing::TestWithParam<std::tuple<bool>> {};
+
+TEST_P(SimBinnedConstrainedTest, ConstrainedAndOffset)
 {
+   using namespace RooFit;
+
+   // parameters
+   const bool parallelLikelihood = std::get<0>(GetParam());
+
+   std::unique_ptr<RooWorkspace> wPtr = makeSimBinnedConstrainedWorkspace();
+   auto &w = *wPtr;
+
+   RooAbsPdf *pdf = w.pdf("model");
+   RooAbsData *data = w.data("data");
+
    // do a minimization, but now using GradMinimizer and its MP version
-   nll.reset(pdf->createNLL(*data, RooFit::Constrain(RooArgSet(*w.var("alpha_bkg_obs_A"))),
-                            RooFit::GlobalObservables(RooArgSet(*w.var("alpha_bkg_obs_B"))), RooFit::Offset(true)));
+   std::unique_ptr<RooAbsReal> nll{pdf->createNLL(*data, Constrain(*w.var("alpha_bkg_obs_A")),
+                                                  GlobalObservables(*w.var("alpha_bkg_obs_B")), Offset(true))};
 
    // parameters
    std::size_t NWorkers = 2;
 
    std::unique_ptr<RooArgSet> values(pdf->getParameters(data));
 
-   values->add(*pdf);
-   values->add(*nll);
```

Comment:
Are you sure this makes no difference? Did you check the values and steps of the ensuing fits? `nll` probably shouldn't make a difference, because it is not used again. `pdf` will probably be recalculated because the parameters will be reset. Nothing we're missing?

Reply:
Right, `nll` is not used, ans also `pdf` will be recalculated anyway because of the dirty state propagation from the changed parameters. So setting the cached value of a PDF via `RooAbsCollection::assign()` has no effect (unless you would change the operation mode of the `RooAbsArg`, but that's not done in the test).